### ISSUE TITLE
[COGNITION]: Left Nav - In this section link SHOULD have an accessible label

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#8992]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️  Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The **In this section** button is using `aria-describedby` to point at the UL for an accessible label. Because the UL doesn't have a heading or descriptive text, the button won't provide any further information to screen readers. It would be better to adjust this to use `aria-label` that includes the visual text and offers some more meaning. Screen shot and code snippets below.

## Point of Contact

**VFS Point of Contact:** _Jennifer_

## Environment

* https://staging.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/ (or any other page at mobile size with the "In this section" menu

### Current code
```diff
<button
- aria-describedby="va-sidenav-ul-container"
  class="medium-screen:vads-u-display--none va-sidenav-default-trigger vads-u-color--primary"
  type="button">
    In this section
-   <i class="fa fa-bars"></i>
</button>
```

### Proposed code
```diff
<button
+ aria-label="Pages in this section"
  class="medium-screen:vads-u-display--none va-sidenav-default-trigger vads-u-color--primary"
  type="button">
    In this section
+   <i aria-hidden="true" class="fa fa-bars" role="presentation"></i>
</button>
```

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2020-05-12 at 9 54 40 AM](https://user-images.githubusercontent.com/934879/81709945-e7b91500-9437-11ea-8654-92d3ac78534b.png)

**Comment by mpelzsherman** [comment]
Question for @1Copenut:

With VoiceOver active on my iPhone, the current implementation reads " In This Section. Button. Secondary Navigation. Landmark. Services and location, News and events, About VA Pittsburgh."

The suggested implementation reads "Pages In This Section. Button. Secondary Navigation. Landmark." The collapsed <li> menu items in the <ul> are not read.

Can you help me understand why the proposed change is better?

**Comment by jenstrickland** [comment]
When I test the recommended fix it behaves the same as the first implementation, reads the collapsed menu items, with the enhancement of changing "In this section" to "Pages in this section" which is more explicit for screen reader users. 

This led to discovering another issue with the implementation:
This .va-sidenav-wrapper has a height of 45px with overflow:hidden, so the screen reader moves through the items that should be hidden. Normally, the `ul` would have display:none and the button would toggle the expand/collapse.

**Comment by mpelzsherman** [comment]
Thanks @jenstrickland . Can you please clarify whether we want the LI items (Services and location, News and events, About VA Pittsburgh) to be read when the "In this section" link is selected? The suggested fix will prevent that, and it sounds like you are saying this is in fact preferable to the current behavior. (?)

Interesting note about the collapse/expand implementation. Good to know, thanks!

**Comment by jenstrickland** [comment]
The desired behavior is that the list items should not be read unless the button expands what should be hidden items. 

As it is now, if the screen reader reads through the sidenav past the "In this section" the user may get stuck on the children list items. I got stuck and it took me a bit to figure out how to get out of it. I'm going to write up another ticket to request the implementation be changed, and add it to the same epic. cc @1Copenut

**Comment by jenstrickland** [comment]
Ticketed the keyboard/focus trap: #10997

**Comment by mpelzsherman** [comment]
PR with the proposed code change is [here](https://github.com/department-of-veterans-affairs/vets-website/pull/13398).

**Comment by mmiddaugh** [comment]
This is ready for your review, @jenstrickland

**Comment by jenstrickland** [comment]
Validated. 🎉
